ocf/ocf.go

```diff
@@ -241,12 +286,36 @@ type Encoder struct {
 // If the writer is an existing ocf file, it will append data using the
 // existing schema.
 func NewEncoder(s string, w io.Writer, opts ...EncoderFunc) (*Encoder, error) {
+	return newEncoder(
+		func(cache *avro.SchemaCache) (avro.Schema, error) {
```

Comment:
There is no real need for `newEncoder`. This function can parse the scheme and call `NewEncoderWithSchema` which would simplify everything.

Reply:
If needs to process options in order to do that, so it can use the correct schema cache to parse it. So I did this to avoid process options 2x. Alternatively, there could still be an unexported `newEncoder` that takes the parsed `avro.Schema` _and_ accepts options as `*encoderConfig` instead of `...EncoderFunc`. Preference?

Reply:
I made a change to this. See what you think.

Reply:
Looks much better, thanks.